Arm/Cfg/Cfg.lean

```diff
@@ -88,25 +94,31 @@ deriving Repr
 
 instance : ToString LoopsInfo where toString li := toString (repr li)
 
-/-- CFG collects all the control-flow information gleaned from the
-program during static analysis. -/
+/--
+CFG collects all the control-flow information gleaned from the
+program during static analysis.
+-/
 structure Cfg where
-  start_address : BitVec 64
-  graph         : ForwardGraph  := #[]
-  loops_info    : LoopsInfo := #[]
+  start_address       : BitVec 64
+  graph               : ForwardGraph  := #[]
+  loops_info          : LoopsInfo := #[]
+  maybe_modified_regs : Array RegType := #[]
 deriving Repr
 
 instance : ToString Cfg where toString cfg := toString (repr cfg)
 
-/-- We can detect a loop if we find an entry where some `to_pc` is
+/--
+We can detect a loop if we find an entry where some `to_pc` is
 less than its corresponding `from_pc`, i.e., there is a back-jump from
 `from_pc` to `to_pc`.
 
 In that case, `to_pc` has the loop target instruction and `from_pc`
 has the loop guard instruction. The first instruction that will be
 executed after the loop terminates will also be a member of `to_insts`
-of type `Seq`. -/
-private def loop_detected (from_inst : InstType) (to_insts : List InstType) : IO (Option LoopInfo) :=
+of type `Seq`.
+-/
+private def loop_detected (from_inst : InstType) (to_insts : List InstType) :
+  Except String (Option LoopInfo) :=
```

Comment:
```suggestion
  Except Format (Option LoopInfo) :=
```

If we use a `Format`, we can write richer, structured logs with combinators like `nest, group`. We can then call `toString : Format -> String` on a `Format` to make a string out of it.